walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,91 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.LinkedList;
 
 public final class FindMeetingQuery {
+
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+
+    // Using linked list as removing the first element will take O(1) time
+    LinkedList<Event> eventList = new LinkedList<Event>(events);
+    // Collections.sort() -> O(nlogn)
+    Collections.sort(eventList, Event.ORDER_EVENT_BY_START); 
+
+    ArrayList<TimeRange> s = new ArrayList<TimeRange>();
+
+    long currentTime = (long) TimeRange.START_OF_DAY;
+
+    // making sure the request is within bounds
+    if (request.getDuration() > TimeRange.WHOLE_DAY.duration()) {
+      return s;
+    }
+
+    if (request.getDuration() < 0) {
+      return s;
+    }
+    while (currentTime < (long) TimeRange.END_OF_DAY) {
```

Comment:
Please move this loop into its own helper function that just returns the list of events. Consider splitting the if blocks in this loop into their own helper functions where applicable.